chia/server/ws_connection.py

```diff
@@ -202,13 +202,22 @@ async def perform_handshake(
         self.outbound_task = asyncio.create_task(self.outbound_handler())
         self.inbound_task = asyncio.create_task(self.inbound_handler())
 
+    def do_close_callback(self, ban_time: int, closed_connection: bool = False) -> None:
+        try:
+            self.close_callback(self, ban_time, closed_connection)
+        except Exception:
+            error_stack = traceback.format_exc()
+            self.log.error(f"Error calling close callback: {error_stack}")
+
```

Comment:
I think this layer may be able to be eliminated by using the `chia.util.log_exceptions.log_exceptions()` context manager instead.  Almost nobody knows about this and I keep forgetting about it too.  I should go on a quest and apply it as much as possible.  Maybe.

https://github.com/Chia-Network/chia-blockchain/blob/0befdec071f49708e26c7638656874492c52600a/chia/util/log_exceptions.py

Anyways, with that out of the way I think we could drop the `closed_connection` parameter and just have a ban function and a close function.

Reply:
Can you pseudo code this out how it might look? Not sure I follow

Reply:
I think my problem is the "close_callback" is implemented in server.py, which manages the connection list and the banned list - so this would require 2 callbacks into server.py?

Reply:
I removed this `do_close_callback` helper function and use log_exceptions context manager